### PR TITLE
FreeBSD/NetBSD compilation issues

### PR DESCRIPTION
**Pull request by ton31337**
Descriptions in commits.

### CODE REVIEW

File: compat/compat.c [hunk 1]
```diff
@@ -265,7 +265,18 @@ realpath(const char *path, char *resolved_path)
...
     return resolved;
 }
+#elif defined (__NetBSD__)
+char *
+realpath(const char *path, char *resolved_path)
+{
+    ssize_t nbytes;
 
+    nbytes = readlink(path, resolved_path, PATH_MAX);
+    if (nbytes == -1)
```
Comment (michalvasko) [inline compat/compat.c:275]: We use `{}` even for one-line `if`s.

**Review by michalvasko — CHANGES_REQUESTED** [review]
Formatting test failed and I found at least one problem.

File: compat/compat.c (hunk 1 above)
Comment (ton31337) [reply compat/compat.c]: Thanks, fixed.

**Comment by michalvasko** [comment]
I ran `uncrustify` on the code locally, there is still a difference
```diff
diff --git a/compat/compat.c b/compat/compat.c
index aa291902..cf0ca87e 100644
--- a/compat/compat.c
+++ b/compat/compat.c
@@ -265,6 +265,7 @@ realpath(const char *path, char *resolved_path)
     }
     return resolved;
 }
+
 #elif defined (__NetBSD__)
 char *
 realpath(const char *path, char *resolved_path)
@@ -277,6 +278,7 @@ realpath(const char *path, char *resolved_path)
     }
     return resolved_path;
 }
+
 #else
 #error No realpath() implementation for this platform is available.
 #endif
```
so please just add those newlines and the tests should pass.

**Comment by ton31337** [comment]
> I ran `uncrustify` on the code locally, there is still a difference
> 
> ```diff
> diff --git a/compat/compat.c b/compat/compat.c
> index aa291902..cf0ca87e 100644
> --- a/compat/compat.c
> +++ b/compat/compat.c
> @@ -265,6 +265,7 @@ realpath(const char *path, char *resolved_path)
>      }
>      return resolved;
>  }
> +
>  #elif defined (__NetBSD__)
>  char *
>  realpath(const char *path, char *resolved_path)
> @@ -277,6 +278,7 @@ realpath(const char *path, char *resolved_path)
>      }
>      return resolved_path;
>  }
> +
>  #else
>  #error No realpath() implementation for this platform is available.
>  #endif
> ```
> 
> so please just add those newlines and the tests should pass.

Thanks, fixed.

**Review by michalvasko — APPROVED** [review]

**Pull request merged**